### ISSUE TITLE
Muted group chats keep being un-muted

### ISSUE THREAD

**Issue by k-bx** [signalapp/Signal-Android#13535]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description

I have a heavily-used Signal account with several groups having hundreds of messages daily, sometimes more. I put them all on "mute forever". When I open the tablet after a day or two, I see these groups being un-muted again. Muting again un-mutes eventually.

### Steps to reproduce

- mute group chat forever
- wait
- see it unmuted eventually

**Actual result:** Group is unmuted and notifications are shown for every message
**Expected result:** Group is muted and no notification is shown

### Device info

**Device:** Galaxy Tab S9+ 5G
**Android version:** 14
**Signal version:** 7.4.2

**Comment by cody-signal** [comment]
Are you muting the chat on a linked device or on the android device? It looks like the app is struggling with a backlog of messages that can really gum things up. Things seem extra slow for some reason.

If you are able to let the device fully catch up, and then mute the chat from Android and confirm it's muted on linked devices that should do the trick. But it's something we need to look into for sure regardless

**Comment by k-bx** [comment]
@cody-signal primary device. There are several linked devices but I don't do anything on them.

Yes, the chats are heavy. If I mute on a caught-up env, it seems to hold for longer, but eventually it'll still un-mute.

**Comment by cody-signal** [comment]
Thanks, if you aren't doing much with the linked devices or don't need them. You can try unlinking them for a bit and see if any behavior changes. My hunch, is that one of the linked devices is also struggling to catch up and changes the mute status of the chat while it does so because of a race condition with storage service. I'd be curious if you've ever confirmed a chat is muted on all of your caught up devices at the same time?

Should you have to do it? No. But if you could, and see, then would help focus our search.

**Comment by k-bx** [comment]
I need my linked devices very much, they read messages programmatically via https://github.com/AsamK/signal-cli but don't write anything, only using the "receive messages" api.

Should I raise an issue in signal-cli ?

**Comment by cody-signal** [comment]
Ahh, I see.

**Comment by k-bx** [comment]
@cody-signal do you think I should ping the signal-cli authors? Are there any hints on what exactly is causing this?

**Comment by codywohlers** [comment]
I have a similar issue where signal-cli is the main device and some chats get unmuted on the linked devices when signal-cli is running.  https://github.com/AsamK/signal-cli/discussions/1537

**Comment by farewelltospring** [comment]
https://github.com/AsamK/signal-cli/issues/1651 may be relevant here too.

**Comment by winkelnp** [comment]
This issue recently popped up for me again, now muted chats don't stay muted longer than 3s, no matter what duration I choose, did anyone ever find out what was happening here?

**Comment by dfoxfranke** [comment]
> This issue recently popped up for me again, now muted chats don't stay muted longer than 3s, it seems only if I choose "forever" as duration, the rest seems to work, did anyone ever find out what was happening here?

Same here, starting just now.

**Comment by Ged-fi** [comment]
I'm seeing the same - mute "forever" just stopped working. Have confirmation from other users, too, that it's now broken.

Android version 7.31.1

**Comment by lmaxsmith** [comment]
Same

**Comment by pjjanak** [comment]
Started happening to me yesterday as well. No signal-cli set up or anything. Just my android as the primary device and they desktop apps as secondary.

**Comment by EnGamma** [comment]
Just started for me for the first time.   First noticed on Windows desktop app and then on Android phone.   Groups I had muted are now spamming me.  The mute setting seems to take, but a few seconds later it turns off.  I've tried setting on phone and on desktop apps.   Not working on any.  I had not experienced this before today.

**Comment by EnGamma** [comment]
Update:

I suspected an interaction among Android and Desktop App (open on two computers).   I closed the Signal desktop app on the two computers (ensured not minimized) and then tried to reset Mute Always on Android phone.   That seems to be working.

**Comment by winkelnp** [comment]
> Update:
> 
> I suspected an interaction among Android and Desktop App (open on two computers). I closed the Signal desktop app on the two computers (ensured not minimized) and then tried to reset Mute Always on Android phone. That seems to be working.

For me as soon as I then reopen the desktop app it unmutes the chats again

**Comment by EnGamma** [comment]
> For me as soon as I then reopen the desktop app it unmutes the chats again

@winkelnp are you sure that the desktop apps weren't minimized to the Windows System Tray area on taskbar (i.e., still active)?

**Comment by winkelnp** [comment]
> are you sure that the desktop apps weren't minimized to the Windows System Tray area on taskbar (i.e., still active)?

yes I am, though it does reopen into the SysTray, though I doubt that makes a difference
(for the record, I killed the process, checked it wasn't running, then started it again)

I was on v7.40.0, just updated to v7.40.1 though and that seems to have fixed it for me too (at least it doesn't immediately unmute, any more I guess we'll see)

**Comment by winkelnp** [comment]
Looks like this was fixed by 2001e4d, so this issue should be closable now

**Comment by ccoenen** [comment]
@no-usernames-left As mentioned in the post directly above yours: the problem stems from Signal-Desktop. Update that, and then the setting shouldn't get reset again.

**Comment by kuflierl** [comment]
I suspect this is the reason for https://github.com/mollyim/mollyim-android/issues/435

**Comment by manavnagpal08** [comment]
Try with different API it may be used as null value . That cause mute
button forever only image icon of muted is showing but it may be unmuted .
If first was not a solution try changing the unmuted icon image… or the
name properly of every icon .in defining
